RealmSwift/Logger.swift

```diff
@@ -0,0 +1,275 @@
+         print("Realm Log - \(level): \(message)")
+     })
+     logger.log(level: .info, message: "Info DB: Database opened succesfully")
+     ```
+
+     - parameter level: The log level for the message.
+     - parameter category: The log category for the message.
+     - parameter message: The message to log.
+     */
+    internal func log(level: LogLevel, category: LogCategory = Category.realm, message: String) {
+        self.log(with: level, category: category.toString(), message: message)
+    }
+
+    /**
+     Creates a logger with the associated log level, category and the logic function to define your own logging logic.
+
+     ```swift
+     let logger = Logger(level: .info, category: Category.All, logFunction: { level, category, message in
+         print("\(category.toString()) - \(level): \(message)")
+     })
+     logger.log(level: .info, category: Category.realm, message: "Info DB: Database opened succesfully")
+     ```
+
+     - parameter level: The log level to be set for the logger.
+     - parameter category: The log category to be set for the logger, by default it will setup the top Category `Category.realm`
+     - parameter function: The log function which will be invoked whenever there is a log message.
+
+     - note:By setting the log level of a category, it will set all its subcategories log level as well.
+     - SeeAlso: `LogCategory`
+     */
+    public convenience init(level: LogLevel, category: LogCategory = Category.realm, function: @escaping @Sendable (LogLevel, LogCategory, String) -> Void) {
+        self.init(level: level, category: category.toString()) { level, cat, message in
+            function(level, Category.fromString(cat)!, message)
+        }
+    }
+
+    /**
+     Sets the log level for a given log category.
+
+     - parameter level: The log level to be set for the logger.
+     - parameter category: The log category to be set for the logger, by default it will setup the top Category `Category.realm`
+
+     - note:By setting the log level of a category, it will set all its subcategories log level as well.
+     - SeeAlso: `LogCategory`
+     */
+    public func setLogLevel(_ level: LogLevel, for category: LogCategory = Category.realm) {
+        Logger.shared.__setLevel(level, category: category.toString())
+    }
+
+    /**
+     Gets the current log level of a log category.
+
+     - parameter category: The target log category.
+
+     - returns: The `LogLevel` for the given category.
+     - SeeAlso: `LogCategory`
+     */
+    public func getLogLevel(for category: LogCategory) -> LogLevel {
+        Logger.shared.__getLevelForCategory(category.toString())
+    }
+}
+
+/// Defines a log category for the Realm `Logger`.
+public protocol LogCategory: Sendable {
+    /**
+     Returns the string represtation of the Log category.
+
+     - returns: A string representing the log category.
+     - SeeAlso: `LogCategory`
+     */
+    func toString() -> String
```

Comment:
`toString()` is a very unidiomatic name for this. `var rawValue: String { get }` is the normal name.